mesonbuild/backend/ninjabackend.py

```diff
@@ -664,8 +664,8 @@ def generate(self, capture: bool = False, vslite_ctx: T.Optional[T.Dict] = None)
             self.generate_dist()
             mlog.log_timestamp("Dist generated")
             key = OptionKey('b_coverage')
-            if (key in self.environment.coredata.optstore and
-                    self.environment.coredata.optstore.get_value(key)):
+            if key in self.environment.coredata.optstore and\
+                    self.environment.coredata.optstore.get_value_for('b_coverage'):
```

Comment:
The reason I changed this is just a matter of numbers. I computed the number of `OptionKey`s with the following zsh commands:

```
grep 'OptionKey\(.\w+.\)' mesonbuild/**/*.py |wc
g 'OptionKey\(.\w+.,' mesonbuild/**/*.py |wc
```

The number of cases where `OptionKey` is only used with a string is more than 200 whereas the number of times it takes other arguments is less than 20. Saying "a string only means a default constructed key" cuts down on a lot of boilerplate and makes (IMHO) the code easier to read.

And because all of this is now inside the OptionStore it is easy to validate all incoming data, convert it to keys and do all internal work with keys.